runtime/cudaq/platform/mqpu/MultiQPUPlatform.cpp

```diff
@@ -102,65 +133,107 @@ class MultiQPUQuantumPlatform : public cudaq::quantum_platform {
       return "";
     };
 
-    if (description.find("remote_execution") != std::string::npos) {
-      if (!cudaq::registry::isRegistered<cudaq::QPU>("RemoteSimulatorQPU"))
+    const auto qpuSubType = getQpuType(description);
+    if (!qpuSubType.empty()) {
+      if (!cudaq::registry::isRegistered<cudaq::QPU>(qpuSubType))
         throw std::runtime_error(
-            "Unable to retrieve RemoteSimulatorQPU implementation.");
-      auto urls = cudaq::split(getOpt(description, "url"), ',');
-      auto sims = cudaq::split(getOpt(description, "backend"), ',');
-      // Default to qpp simulator if none provided.
-      if (sims.empty())
-        sims.emplace_back("qpp");
-      // If no URL is provided, default to auto launching one server instance.
-      const bool autoLaunch =
-          description.find("auto_launch") != std::string::npos || urls.empty();
-
-      const auto formatUrl = [](const std::string &url) -> std::string {
-        auto formatted = url;
-        // Default to http:// if none provided.
-        if (!formatted.starts_with("http"))
-          formatted = std::string("http://") + formatted;
-        if (!formatted.empty() && formatted.back() != '/')
-          formatted += '/';
-        return formatted;
-      };
-      if (autoLaunch) {
-        urls.clear();
-        const auto numInstanceStr = getOpt(description, "auto_launch");
-        // Default to launching one instance if no other setting is available.
-        const int numInstances =
-            numInstanceStr.empty() ? 1 : std::stoi(numInstanceStr);
-        cudaq::info("Auto launch {} REST servers", numInstances);
-        for (int i = 0; i < numInstances; ++i) {
-          m_remoteServers.emplace_back(
-              std::make_unique<cudaq::AutoLaunchRestServerProcess>(i));
-          urls.emplace_back(m_remoteServers.back()->getUrl());
+            fmt::format("Unable to retrieve {} QPU implementation. Please "
+                        "check your installation.",
+                        qpuSubType));
+      if (qpuSubType == "NvcfSimulatorQPU") {
+        platformQPUs.clear();
+        auto simName = getOpt(description, "backend");
+        if (simName.empty())
+          simName = "custatevec-fp32";
+        std::string configStr =
+            fmt::format("target;nvcf;simulator;{}", simName);
+        {
+          auto apiKey = getOpt(description, "api_key");
+          if (!apiKey.empty())
+            configStr += fmt::format(";api_key;{}", apiKey);
+        }
+        {
+          auto functionId = getOpt(description, "function_id");
+          if (!functionId.empty())
+            configStr += fmt::format(";function_id;{}", functionId);
+        }
+        {
+          auto versionId = getOpt(description, "version_id");
+          if (!versionId.empty())
+            configStr += fmt::format(";version_id;{}", versionId);
         }
```

Comment:
Could probably shorten this up a bit. See below for suggestion, which is untested.
```suggestion
        auto getOptAndSetConfig = [&](const std::string &key) {
          auto val = getOpt(description, key);
          if (!val.empty())
            configStr += fmt::format(";{};{}", key, val);
        };
        getOptAndSetConfig("api_key");
        getOptAndSetConfig("function_id");
        getOptAndSetConfig("version_id");
```